### ISSUE TITLE
Instructions: T-junction recognition

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1130]
When currently hitting a T-junction, we don't create a special instruction, just turn left or right.

Here is an [example](https://graphhopper.com/maps/?point=52.336388%2C5.298457&point=52.332625%2C5.300308). Instead of simply creating a "turn right" instruction, we could create an instruction "turn right, at the end of the street". As you can see, about 30m before the actual turn there is another possible turn. When navigating the turn might be ambiguous.

**Comment by karussell** [comment]
IMO this is a low prio thing, as the ambiguity could be there for other non-T-junction cases as well. So this is just a more clear instruction in a few cases (?) and other improvements could be more important

**Comment by boldtrn** [comment]
> So this is just a more clear instruction in a few cases (?) and other improvements could be more important

Yes, this is correct. I don't know many instructions could benefit from this improvement.

The issue here is that most T-junctions are not ambiguous. But T-junctions are quite common. I would say ~5% of the turns, when ignoring motorways, but most of them are not ambiguous. Maybe we need a way to identify ambiguous instructions and improve them instead?
